bayesian/utility/v2/component_analyses.py

```diff
@@ -131,22 +144,28 @@ def get_batch_ca_data(ecosystem: str, packages) -> dict:
     else:
         semver_packages = packages
 
+    graph_data_fetcher = []
     if len(semver_packages) > 0:
-        response = GraphAnalyses.get_batch_ca_data(ecosystem, semver_packages)
+        get_semver_data = functools.partial(GraphAnalyses.get_batch_ca_data, ecosystem)
+        graph_data_fetcher = list(_fetcher_in_batches(get_semver_data, semver_packages))
 
     if len(pseudo_version_packages) > 0:
-        pseudo_response = GraphAnalyses.get_batch_ca_data_for_pseudo_version(
-            ecosystem, pseudo_version_packages)
-        # Merge both data into one.
-        if response:
-            response['result']['data'] += pseudo_response['result']['data']
-        else:
-            response = pseudo_response
+        get_pseudo_data = functools.partial(GraphAnalyses.get_batch_ca_data_for_pseudo_version,
+                                            ecosystem)
+        graph_data_fetcher += list(_fetcher_in_batches(get_pseudo_data, pseudo_version_packages))
+
+    response = {
+        "result": {
+            "data": []
+        }
+    }
+    result = EXECUTOR.map(lambda f: f(), graph_data_fetcher)
```

Comment:
Things looks good by review, but do ample testing with pseudo version + sem-version, only sem-version, only pseudo version use- cases.

Reply:
I guess the existing unit tests covers those scenarios.